### PR TITLE
Feature/tet 874 ingest stova events

### PR DESCRIPTION
**Pull request by DeanElliott96**
### Description of change

Part 2 of 3.

Data is still not being ingested yet but this work sets the functions which run for an ingested file and tests against a test fixture. This work also links the `StovaEvent` to a DataHub `Event`.

The model file was renamed which makes it look like a new file but there a few model changes from CharField to IntegerField and also new functions on the model for saving the `StovaEvent` to a Datahub `Event`.

A Datahub `Event` requires an `EventType`, this field is not providing by Stova so it has been decided to give each `StovaEvent` the `EventType` of 'Stova - unknown event type'. This is filterable on the frontend to find all Stova Events and can also be changed if a user wishes.

Part 3 will be adding a cron task to ingest files from the S3 bucket.

### Checklist

* [x] Has this branch been rebased on top of the current `main` branch?
* [x] Is the CircleCI build passing?

### CODE REVIEW

File: datahub/company_activity/models/stova_events.py
Comment (DeanElliott96) [inline datahub/company_activity/models/stova_events.py:1]: Renamed file

File: datahub/company_activity/models/stova_event.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+from django.conf import settings
+from django.db import models, transaction
+
+from datahub.event.models import Event, EventType
+from datahub.metadata.utils import get_country_by_country_name
+
+MAX_LENGTH = settings.CHAR_FIELD_MAX_LENGTH
+
+
+class StovaEvent(models.Model):
```
[the remark this answers is not in the document]
Comment (DeanElliott96) [reply datahub/company_activity/models/stova_event.py:12]: This matches the ingested file, won't change.

**Review by baarkerlounger — APPROVED** [review]

File: datahub/company_activity/tasks/ingest_stova_events.py [hunk 2]
```diff
@@ -20,7 +20,7 @@ class StovaEventIngestionTask:
     def __init__(self):
         self._existing_ids = []
 
-    def ingest(self, bucket, file):
+    def ingest(self, bucket: str, file: str) -> None:
```
[the remark this answers is not in the document]
Comment (DeanElliott96) [reply datahub/company_activity/tasks/ingest_stova_events.py:23]: I had to changer some values in the class as I was writing the other code with type hints so I added them to the rest of this class as well. I could add the type hints to the rest of the file if you want as well?

File: datahub/metadata/utils.py [hunk 3]
```diff
@@ -30,3 +30,25 @@ def get_latest_exchange_rate():
...
         to_currency_code='GBP',
     ).order_by('-created_on').first().exchange_rate
+
+
+def get_country_by_country_name(name: str, default_iso='') -> Country:
+    """
+    Attempts to match a Country from the given name or by the given default_iso is match by name
+    is not found.
+
+    :param name: a country name from Stova
```
Comment (DeanElliott96) [inline datahub/metadata/utils.py:40]: I should change this docstring now its a util to not say Stova.

File: datahub/company_activity/tasks/ingest_stova_events.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (baarkerlounger) [reply datahub/company_activity/tasks/ingest_stova_events.py:23]: No strong preference really. Just wondering if we should be consistent about using type hints everywhere or not at all. Do they add much value when they're not enforced?

File: datahub/company_activity/tests/test_tasks/fixtures/stova/stovaEventFake2.jsonl.gz
[the remark this answers is not in the document]
Comment (DeanElliott96) [reply datahub/company_activity/tests/test_tasks/fixtures/stova/stovaEventFake2.jsonl.gz:1]: "original file" - The previous file on GitHub that this replaces.

I renamed as its a zip file so git detected no changes when I replaced it. I could rename it back if you think its needed.

File: datahub/company_activity/models/stova_event.py [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+                'name': self.name,
+                'event_type_id': self.get_or_create_stova_event_type().id,
+                'start_date': self.start_date.date() if self.start_date else '',
+                'end_date': self.end_date.date() if self.end_date else '',
+                'address_1': self.location_address1,
+                'address_2': self.location_address2,
+                'address_town': self.location_city,
+                'address_county': self.location_state,
+                'address_postcode': self.location_postcode,
+                'address_country': get_country_by_country_name(self.name, 'GB'),
```
Comment (oliverjwroberts) [inline datahub/company_activity/models/stova_event.py:84]: Should you be passing `self.country` here?

File: datahub/company_activity/tasks/ingest_stova_events.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (DeanElliott96) [reply datahub/company_activity/tasks/ingest_stova_events.py:23]: Maybe one for dev huddle?

I think they add value value even if not enforced as they help document the code.

[the remark this answers is not in the document]
Comment (baarkerlounger) [reply datahub/company_activity/tasks/ingest_stova_events.py:23]: Yeah not intending to hold up this PR or anything just wondering out loud.

**Review by oliverjwroberts — COMMENTED** [review]
One comment from me, but the rest looks good.

File: datahub/company_activity/models/stova_event.py (hunk 4 above)
Comment (DeanElliott96) [reply datahub/company_activity/models/stova_event.py]: Fixed now with test.

**Pull request merged**